Sources/Algorithms/Trim.swift

```diff
@@ -29,8 +84,66 @@ extension BidirectionalCollection {
   public func trimming(
     while predicate: (Element) throws -> Bool
   ) rethrows -> SubSequence {
-    let start = try endOfPrefix(while: predicate)
-    let end = try self[start...].startOfSuffix(while: predicate)
-    return self[start..<end]
+    return try trimmingPrefix(while: predicate).trimmingSuffix(while: predicate)
+  }
+  
+  /// Returns a `SubSequence` formed by discarding all elements at the end
+  /// of the collection which satisfy the given predicate.
+  ///
+  /// This example uses `trimmingSuffix(while:)` to get a substring without the white
+  /// space at the end of the string:
+  ///
+  ///     let myString = "  hello, world  "
+  ///     print(myString.trimmingSuffix(while: \.isWhitespace)) // "  hello, world"
+  ///
+  /// - Parameters:
+  ///    - predicate: A closure which determines if the element should be
+  ///                 omitted from the resulting slice.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of this collection.
+  ///
+  @inlinable
+  public func trimmingSuffix(
+    while predicate: (Element) throws -> Bool
+  ) rethrows -> SubSequence {
+    let end = try startOfSuffix(while: predicate)
+    return self[startIndex..<end]
+  }
+}
+
+//===----------------------------------------------------------------------===//
+// trim(while:) / trimSuffix(while:)
+//===----------------------------------------------------------------------===//
+
+extension BidirectionalCollection where Self: RangeReplaceableCollection {
+  @inlinable
+  public mutating func trim(
+    while predicate: (Element) throws -> Bool
+  ) rethrows {
+    replaceSubrange(startIndex..<endIndex, with: try trimming(while: predicate))
+  }
+  
+  @inlinable
+  public mutating func trimSuffix(
+    while predicate: (Element) throws -> Bool
+  ) rethrows {
+    let start = try startOfSuffix(while: predicate)
+    removeSubrange(start..<endIndex)
+  }
+}
+
+extension BidirectionalCollection where Self == Self.SubSequence {
```

Comment:
These overloads look great! Unfortunately, for collections that meet both requirements (i.e. range-replaceable collections that are also self-slicing), trying to call the method ends up ambiguous. We used to fix this by adding a third overload in an extension with _both_ constraints, so that it would be more specific than the other two, but now we have the `@_disfavoredOverload` attribute. Can you add `@_disfavoredOverload` to the `RangeReplaceableCollection` versions of the methods?